### PR TITLE
Fix reading http4s Uris

### PR DESCRIPTION
**Pull request by jcranky**
Make sure that errors are not ignored and that the values are put in the
correct fields.

### CODE REVIEW

**Comment by ruippeixotog** [comment]
The code isn't compiling on Scala 2.11 because `toTry` is 2.12-only. Can you fix that issue?

**Comment by jcranky** [comment]
@ruippeixotog I didn't notice the build failure, I'll look into that

**Comment by jcranky** [comment]
PR updated :)

**Review by ruippeixotog — APPROVED** [review]
LGTM!

File: modules/http4s/src/test/scala/pureconfig/modules/http4s/Http4sTest.scala [hunk 1]
```diff
@@ -12,6 +12,6 @@ class Http4sTest extends BaseSuite {
   "reading the uri config" should "parse the uri" in {
```
Comment (leifwickland) [inline modules/http4s/src/test/scala/pureconfig/modules/http4s/Http4sTest.scala:13]: Please add a case which tests the unhappy path of a bad URI.

**Review by leifwickland — CHANGES_REQUESTED** [review]

File: modules/http4s/src/test/scala/pureconfig/modules/http4s/Http4sTest.scala (hunk 1 above)
Comment (jcranky) [reply modules/http4s/src/test/scala/pureconfig/modules/http4s/Http4sTest.scala:13]: will do

**Comment by jcranky** [comment]
@leifwickland done

File: modules/http4s/src/test/scala/pureconfig/modules/http4s/Http4sTest.scala (hunk 1 above)
Comment (leifwickland) [reply modules/http4s/src/test/scala/pureconfig/modules/http4s/Http4sTest.scala:13]: @jcranky, Thanks!

**Review by leifwickland — APPROVED** [review]
LGTM! 👍 Thanks, @jcranky

**Pull request merged**